### PR TITLE
Changes to enable browser-specific authentication functionality.

### PR DESCRIPTION
**Pull request by y-polonsk**
These changes are needed to allow console to use IAM authentication functionality.  It includes enabling user-specified profile provider (internal feature) and some changes to utils to separate required browser-specific code.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/).
 
+**Unpublished**
+
+**Added**
+* Added browser authentication functionality in IAM code.
```
Comment (gmfeinberg) [inline CHANGELOG.md:10]: Can you tag this as "Cloud service only?"

File: lib/auth/iam/auth_provider.js [hunk 2]
```diff
@@ -315,6 +324,15 @@ ${HttpConstants.DATE}: ${dateStr}`;
         if (compartment != null) {
             ret[HttpConstants.COMPARTMENT_ID] = compartment;
         }
+
+        //Currently proxy uses presence this header to identify requests from
```
Comment (gmfeinberg) [inline lib/auth/iam/auth_provider.js:328]: This comment does not read like a sentence...

**Review by gmfeinberg — APPROVED** [review]
Approved, but please fix the couple of comments first

**Comment by y-polonsk** [comment]
Ok, i fixed the comments and merged the PR.

**Pull request merged**